apps/payments/next/app/[locale]/[offeringId]/[interval]/new/page.tsx

```diff
@@ -98,50 +98,53 @@ export default async function New({
 
   let redirectToUrl: URL;
   let cart: ResultCart;
+  let couponCode: string | null | undefined;
 
-  if (searchParams.cartId && fxaUid && searchParams.cartVersion) {
-    cart = await updateCartUidAction(
+  if (searchParams.cartId && searchParams.cartVersion) {
+    const {couponCode: fetchedCoupon} = await getCouponAction(
       searchParams.cartId,
       Number(searchParams.cartVersion),
-      fxaUid,
+    );
+    couponCode = fetchedCoupon;
+  }
+  try {
+    cart = await setupCartAction(
+      interval as SubplatInterval,
+      offeringId,
+      taxAddress,
+      undefined,
+      couponCode || coupon,
```

Comment:
nitpick: Could these be renamed to indicate the source? Was thinking something like `cartCoupon` and `searchParamsCoupon` maybe?